content/md/posts/2023-07-08-unit-testing.md

```diff
@@ -0,0 +1,66 @@
+{:title "Unit testing"
+ :layout :post
+ :tags  ["cljs" "testing"]
+ :toc true}
+
+
+## Overview
+
+This article covers what unit testing is, why it is useful, and an example of unit testing in a project.
+
+## What is unit testing?
+
+ Unit testing is about testing a function, using the inputs and verifying the outputs. Unlike acceptance testing which tests the result of a collection of functions interacting together, unit testing just grabs one function and sees whether it reacts correctly given certain inputs.
+ 
+
+## Why is it useful?
+ 
+One possible situation in where you could have a unit test is if you are testing the registration feature for a website. What could be tested is that given the information for a user, registering a user with that information should result in a new user appearing in the database. That user's information should match the same information provided when registering earlier.
```

Comment:
Is not clear to me why a `unit test` .... the second paragraph sounds a bit more like acceptance testing ... perhaps explaining first the scenario. would be a better start ... or maybe rephrase it a bit

Reply:
I did not realize that my unit tests were actually integration tests. I'll change the scenario so it actually sounds like a unit test.